### PR TITLE
Fuzzing and Scan Build Fixes

### PR DESCRIPTION
**Pull request by ejohnstown**
1. Fuzzing found a bad read when processing an X.509 certificate chain. Use the correct bounds when checking value reads.
2. Fix a potential NULL dereference with the ssh-agent. Move the check inside an everything-ok check.
3. Fix a couple dead writes.

(ZD 14803)

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -3301,31 +3301,29 @@ static int ParseAndVerifyCert(WOLFSSH* ssh, byte* in, word32 inSz,
...
+        /* Get the cert count */
+        ret = GetUint32(&certCount, in, inSz, &m);
     }
 
     if (ret == WS_SUCCESS) {
         word32 count;
 
-        certPt = in + m;
-        m = 0;
+        WLOG(WS_LOG_INFO, "Peer sent certificate count of %d", certCount);
```
Comment (haydenroche5) [inline src/internal.c:3320]: certCount is unsigned int so %d should be %u.

File: src/internal.c [hunk 2]
```diff
@@ -3301,31 +3301,29 @@ static int ParseAndVerifyCert(WOLFSSH* ssh, byte* in, word32 inSz,
...
+        certChain = in + m;
+
         for (count = certCount; count > 0; count--) {
             word32 certSz = 0;
 
-            ret = GetSize(&certSz, certPt, inSz, &m);
+            ret = GetSize(&certSz, in, inSz, &m);
             WLOG(WS_LOG_INFO, "Adding certificate size %d", certSz);
```
Comment (haydenroche5) [inline src/internal.c:3327]: Same issue with %d here. Would also prefer this WLOG get moved after the break on ret != WS_SUCCESS block.

**Review by haydenroche5 — CHANGES_REQUESTED** [review]

File: src/internal.c (hunk 1 above)
Comment (ejohnstown) [reply src/internal.c]: got it

File: src/internal.c (hunk 2 above)
Comment (ejohnstown) [reply src/internal.c]: got it

**Review by haydenroche5 — APPROVED** [review]

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**